cogs/gatekeep/gatekeep.py

```diff
@@ -355,7 +350,112 @@ async def listWords(self, ctx: Context):
+        """
+
+        def check(msg: discord.Message):
+            return msg.author == ctx.author and msg.channel == ctx.channel
+
+        # Get confirmation before initializing. It loops through every member in a server, so it could take a while
+        await ctx.send(warning("This operation may take a while. Type 'yes' to confirm."))
+        try:
+            response = await self.bot.wait_for("message", timeout=30.0, check=check)
+        except asyncio.TimeoutError:
+            await ctx.send("No response after 30 seconds, this operation will not be executed.")
+            return
+
+        if response.content.lower() != "yes":
+            await ctx.send("This operation will not be executed.")
+            return
+
+        # Confirmed, so initialization process begins
+        start = time.time()
+        current = datetime.now(timezone.utc)
+        nDays = await self.config.guild(ctx.guild).get_attr(KEY_NEW_USER_DAYS)()
+        watchList = []
+        for member in ctx.guild.members:
+            # If a member has been in the server for less than X days, then they get added to the watch list (X is configurable)
+            if (
+                current - member.joined_at < timedelta(days=nDays)
+                and not member.guild_permissions.administrator
+                and not await self.bot.is_automod_immune(member)
+            ):
+                watchList.append(int(member.id))
+                self.logger.info(
+                    "%s#%s (%s) added to the watch list.",
+                    member.name,
+                    member.discriminator,
+                    member.id,
+                )
+        await self.config.guild(ctx.guild).get_attr(KEY_WATCH_LIST).set(watchList)
+
+        end = time.time() - start
+        await ctx.send(f"Operation took {end:.3f} seconds.")
+
+    @user.command(name="add")
+    @commands.guild_only()
+    @checks.mod_or_permissions(administrator=True)
+    async def addUser(self, ctx: Context, user: discord.User):
+        """Add a user to the watch list.
+
+        Parameters:
+        -----------
+        user: User
+            The user to be added to the watch list. This can be their username or ID.
+        """
+
+        watchList = await self.config.guild(ctx.guild).get_attr(KEY_WATCH_LIST)()
+        if user.id not in watchList:
+            watchList.append(int(user.id))
+            await self.config.guild(ctx.guild).get_attr(KEY_WATCH_LIST).set(watchList)
+            await ctx.send(f"Added user ID `{user.id}` to the watch list.")
+
+            self.logger.info(
+                "%s#%s (%s) added user ID %s to the watch list for %s.",
+                ctx.author.name,
+                ctx.author.discriminator,
+                ctx.author.id,
+                user.id,
+                ctx.guild.name,
+            )
+        else:
+            await ctx.send(f"User ID `{user.id}` is already in the watch list.")
```

Comment:
Here and in removeUser: this is better suited for `async with ...` since you can directly append and skip the `set`.